tools/build-release-branch.sh

```diff
@@ -16,117 +17,133 @@
-    else
-        echo ""
-        echo "Creating new unbuilt branch $NEW_UNBUILT_BRANCH from current master branch..."
-        echo ""
-        # reset --hard to remote master in case they have local commits in their repo
-        git checkout master && git pull && git reset --hard origin/master
-
-        # Create new branch, push to repo
-        git checkout -b $NEW_UNBUILT_BRANCH
-
-        git push -u origin $NEW_UNBUILT_BRANCH
-        echo ""
-        echo "$NEW_UNBUILT_BRANCH created."
-        echo ""
-        # Verify you want a built version
-        read -n1 -p "Would you like to create a built version of $NEW_UNBUILT_BRANCH as new $NEW_BUILT_BRANCH? [y/N]" reply
-        if [[ 'y' == $reply || 'Y' == $reply ]]
-        then
-            # make sure we're still checked out on the right branch
-            git checkout $NEW_UNBUILT_BRANCH
-
-            git checkout -b $NEW_BUILT_BRANCH
-
-            # New .gitignore for release branches
-            echo ""
-            echo "Creating new .gitignore"
-            echo ""
-            create_release_gitignore
-
-            # Remove stuff from svnignore
-            modify_svnignore
-
-            git checkout $NEW_UNBUILT_BRANCH
-
-            git push -u origin $NEW_BUILT_BRANCH
-
-            # Script will continue on to actually build the plugin onto this new branch...
-        else
-            # Nothing left to do...
-            echo ""
-            echo "Ok, all done then."
-            exit 1
-        fi
-    fi
+	# Prompt for version number.
+	read -p "What version are you releasing? Please write in x.x syntax. Example: 4.9 - " version
+
+	# Declare the new branch names.
+	NEW_UNBUILT_BRANCH="branch-$version"
+	NEW_BUILT_BRANCH="branch-$version-built"
+
+	# Check if branch already exists, if not, create new branch named "branch-x.x"
+	if [[ -n $( git branch -r | grep "$NEW_UNBUILT_BRANCH" ) ]]; then
+		echo "$NEW_UNBUILT_BRANCH already exists.  Exiting..."
+		exit 1
+	else
+		echo ""
+		echo "Creating new unbuilt branch $NEW_UNBUILT_BRANCH from current master branch..."
+		echo ""
+		# reset --hard to remote master in case they have local commits in their repo
+		git checkout master && git pull && git reset --hard origin/master
+
+		# Create new branch, push to repo
+		git checkout -b $NEW_UNBUILT_BRANCH
+
+		git push -u origin $NEW_UNBUILT_BRANCH
+		echo ""
+		echo "$NEW_UNBUILT_BRANCH created."
+		echo ""
+		# Verify you want a built version
+		read -n1 -p "Would you like to create a built version of $NEW_UNBUILT_BRANCH as new $NEW_BUILT_BRANCH? [y/N]" reply
+		if [[ 'y' == $reply || 'Y' == $reply ]]; then
+			# make sure we're still checked out on the right branch
+			git checkout $NEW_UNBUILT_BRANCH
+
+			git checkout -b $NEW_BUILT_BRANCH
+
+			# New .gitignore for release branches
+			echo ""
+			echo "Creating new .gitignore"
+			echo ""
+			create_release_gitignore
+
+			# Remove stuff from svnignore
+			modify_svnignore
+
+			git rm -r --cached .
+			git add .
```

Comment:
I don't think this is ideal. While it works, we end up committing the updated Jetpack dashboard build on this step instead of at the next one. It also has the potential to be quite dangerous. For example, if you are using a custom version of the Gutenberg blocks locally (after running `npx lerna run build --stream --scope='@automattic/jetpack-blocks'` for example), those changes will be committed to the release branch. We don't want that to happen, as the blocks should always be the ones that have been tagged in npm for now.

Reply:
Would you prefer something like the following, mirroring the lines above?

```suggestion
git rm -fg \
	extensions/**/*.css \
	extensions/**/*.gif \
	extensions/**/*.jpeg \
	extensions/**/*.jpg \
	extensions/**/*.js \
	extensions/**/*.json \
	extensions/**/*.jsx \
	extensions/**/*.md \
	extensions/**/*.png \
	extensions/**/*.sass \
	extensions/**/*.scss \
	extensions/**/*.svg \
	**/__snapshots__
# no `git add .` required
```

Reply:
Yeah, maybe removing them like that would be better, since it wouldn't impact any other code by mistake?

Reply:
It seems it helped, but I now see some .js files still committed to the release branch:

![image](https://user-images.githubusercontent.com/426388/54930547-bcaff100-4f17-11e9-90fa-a4405e7023d7.png)

Did it work for you?

Reply:
We had a long chat about this and we're going to accept the source files in the built branches for now as it matches what happens with `_inc/client` right now. We're going to offload stripping these to the deploy-to-svn and .svnignore convention.

Fixing this up in the git built branch is a later thing to tackle.